tests/test_cli_main.py

```diff
@@ -202,6 +214,142 @@ def test_handle_url_with_json_and_stream_url(self, console, args):
             self.assertEqual(console.exit.mock_calls, [call("The stream specified cannot be translated to a URL")])
             console.exit.mock_calls.clear()
 
+    def test_create_filename_invalid_chars_native_rules_auto(self):
+        streamlink_cli.main.args = args = Mock()
+        args.fs_safe_rules = None
+
+        if is_win32:
+            test_chars = INVALID_FILENAME_CHARS["windows"]
+        else:
+            test_chars = INVALID_FILENAME_CHARS["unix"]
```

Comment:
Don't branch in each individual test and create OS-specific tests instead with `skip` decorators. Tests should include as few branches as possible, hence the 100% coverage requirement. It's probably also a good idea to create special test classes for this instead of adding more methods to `TestCLIMain`.